### ISSUE TITLE
Arduino resets the previous state

### ISSUE THREAD

**Issue by aliassad** [kurbatov/firmata4j#12]
when i send digital message to arduino setdititalpin 8 high its work but when i send setdititalpin 9 high
the pin 8 goes low.
The arduino resets its self before sending a new message.

**Comment by kurbatov** [comment]
Which version of firmata (.ino scetch) and firmata4j do you use?
Please, provide a code snippet to reproduce the issue.

**Comment by aliassad** [comment]
Latest standardfirmata.ino and firmata4j 2.3.4

**Comment by aliassad** [comment]
device.getPin(8).sevalue(1);
device.getPin(9).sevalue(1);

//the pin 9 gets high only

**Comment by kurbatov** [comment]
Are those pins in output mode?

**Comment by aliassad** [comment]
yes

**Comment by kurbatov** [comment]
BTW, your snippet is not compilable.
Could you please, give some comprehensive snippet right from your code, where it is visible how do you know that something went wrong. Otherwise it's hardly possible to resolve the issue.

**Comment by aliassad** [comment]
try{
             MainGUI.PORT_NAME=requestPort();
             
             device = new FirmataDevice(MainGUI.PORT_NAME); // construct the Firmata device instance using the name of a port
             device.start(); // initiate communication to the device
             device.ensureInitializationIsDone(); // wait for initialization is done        

             Pin p1=device.getPin(8);
             Pin p2=device.getPin(13);
             
                try {
         for(int i = 0; i < 1000; i++) {
            p1.setValue(1);
            Thread.sleep(500);
            p2.setValue(1);
            Thread.sleep(500);
             p1.setValue(0);
             Thread.sleep(500);
            p2.setValue(0);
            Thread.sleep(500);
         }
      } catch (IllegalStateException | IOException | InterruptedException e) {
         // TODO Auto-generated catch block
         e.printStackTrace();
      }

**Comment by aliassad** [comment]
arduino  resets after each intstruction.

**Comment by kurbatov** [comment]
Try to set `output` mode to the pins explicitly.

```java
    ...
    Pin p1=device.getPin(8);
    Pin p2=device.getPin(13);
    p1.setMode(Pin.Mode.OUTPUT);
    p2.setMode(Pin.Mode.OUTPUT);
    ...
```

**Comment by aliassad** [comment]
Same Issue

**Comment by aliassad** [comment]
I had attached LEDs on Arduino pins. the pin8 led gets low when the pin13 is high.

**Comment by soundanalogous** [comment]
It appears the value of the port is not being tracked across writes. I've produced a crude (I don't write much Java) example to illustrate this:

```Java
package com.testportwrite;

import org.firmata4j.IODevice;
import org.firmata4j.firmata.FirmataDevice;
import org.firmata4j.Pin;

import java.io.IOException;

public class Main {

    public static void main(String[] args) {
        final IODevice device = new FirmataDevice("/dev/cu.usbmodem14231");
        try {
            device.start();
        } catch (IOException ex) {
            throw new RuntimeException(ex);
        }
        try {
            device.ensureInitializationIsDone();
        } catch (InterruptedException e) {
            e.printStackTrace();
            System.exit(1);
        }

        Pin pinA = device.getPin(8);
        Pin pinB = device.getPin(9);

        try {
            pinA.setValue(1);
            pinB.setValue(1);
            try {
                // keep LEDs on for 4 seconds
                Thread.sleep(4000);
            } catch (InterruptedException ex) {
                throw new RuntimeException(ex);
            }
            System.out.println("should have set pins");
        } catch (IOException ex) {
            throw new RuntimeException(ex);
        }

        try {
            device.stop();
        } catch (IOException ex) {
            throw new RuntimeException(ex);
        }

    }
}
```

The call to pinB.setValue(1) will set pinA back to 0. However, the port value should be maintained.

**Comment by kurbatov** [comment]
I have released fixed version 2.3.4.1
